ADONET/AdoNetFundamentals/DLL/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,165 @@
+
+        public Order MapOrder(SqlDataReader reader)
+        {
+            return new Order
+            {
+                Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                Status = reader.GetString(reader.GetOrdinal("Status")),
+                CreatedDate = reader.GetDateTime(reader.GetOrdinal("CreatedDate")),
+                UpdatedDate = reader.GetDateTime(reader.GetOrdinal("UpdatedDate")),
+                ProductId = reader.GetInt32(reader.GetOrdinal("ProductId")),
+            };
+        }
+
+        public void BulkDeleteOrders(string status = null, int month = 0, int year = 0, int productId = 0)
+        {
+            var ordersToDelete = FetchOrdersByFilter(status, month, year, productId);
+
+            if (ordersToDelete.Count > 0)
+            {
+                using (var connection = new SqlConnection(AdoNetDbContext.ConnectionString))
+                {
+                    foreach (var order in ordersToDelete)
+                    {
+                        string deleteQuery = "DELETE FROM Orders WHERE Id = @Id";
+                        connection.Open();
+                        using (var command = new SqlCommand(deleteQuery, connection))
+                        {
+                            command.Parameters.AddWithValue("@Id", order.Id);
+                            command.ExecuteNonQuery();
+                        }
+                        connection.Close();
+                    }
+                }
+            }
+        }
+
+        public Order Create(Order order)
+        {
+            string query = "INSERT INTO Orders " +
+                "(Status, CreatedDate, UpdatedDate, ProductId)" +
+                "VALUES (@Status, @CreatedDate, @UpdatedDate, @ProductId)";
+
+            using (var command = GetCommand(query))
+            {
+                command.Parameters.AddWithValue("@Status", order.Status);
+                command.Parameters.AddWithValue("@CreatedDate", order.CreatedDate);
+                command.Parameters.AddWithValue("@UpdatedDate", order.UpdatedDate);
+                command.Parameters.AddWithValue("@ProductId", order.ProductId);
+
+                command.ExecuteNonQuery();
+            }
+
+            return order;
+        }
+
+        public void Delete(int id)
+        {
+            string query = "DELETE FROM Orders WHERE Id = @Id";
+
+            using (var command = GetCommand(query))
+            {
+                command.Parameters.AddWithValue("@Id", id);
+                command.ExecuteNonQuery();
+            }
+        }
+
+        public List<Order> FetchOrdersByFilter(string status = null, int month = 0, int year = 0, int productId = 0)
+        {
+            var orders = new List<Order>();
+            string query = "SELECT * FROM Orders";
+
+            if (!string.IsNullOrEmpty(status) || month > 0 || year > 0 || productId > 0)
+            {
+                query += " WHERE";
+                List<string> conditions = new List<string>();
```

Comment:
It would be more efficient to use [StringBuilder](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-8.0) for string operations, like concatenation, replacing etc, especially with some conditions. Let's use this instead, please